### PR TITLE
Bidirectional language support

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-1537)
[designs as per this task](https://uktrade.atlassian.net/browse/ED-1506)

# Non-bidirectional language

image

# Bidirectional language
(imagine the actual text is arabic)
image

### CODE REVIEW

File: enrolment/templates/bidi/browse-companies.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{% load trans from i18n %}
```
Comment (konradko) [inline enrolment/templates/bidi/browse-companies.html]: why not call the folder `bidirectional`? Unless `bidi` is a common short form.

File: enrolment/templates/govuk_layout.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 <!DOCTYPE html>
 <!--[if lt IE 9]><html class="lte-ie8" lang="en"><![endif]-->
-<!--[if gt IE 8]><!--><html lang="en"><!--<![endif]-->
+<!--[if gt IE 8]><!--><html lang="en" {% if LANGUAGE_BIDI %}dir="rtl"{% endif %}><!--<![endif]-->
```
Comment (konradko) [inline enrolment/templates/govuk_layout.html]: this is quite cryptic

File: enrolment/tests/test_views.py [hunk 3]
```diff
@@ -336,3 +338,33 @@ def test_international_landing_view_translations(client):
 
     assert response.status_code == http.client.OK
     assert 'language_switcher' in response.context_data
+
+
+def test_conditional_translate_bidi_template(rf):
+    class View(ConditionalEnableTranslationsMixin, TemplateView):
+        template_name_bidi = 'bidi.html'
+        template_name = 'non-bidi.html'
```
Comment (konradko) [inline enrolment/tests/test_views.py]: is `bidi` right-to-left and `non-bidi` left-to-right?

File: enrolment/views.py [hunk 4]
```diff
@@ -21,9 +23,10 @@
 zenpy_client = Zenpy(timeout=5, **ZENPY_CREDENTIALS)
 
 
-class ConditionalEnableTranslationsMixin:
+class ConditionalEnableTranslationsMixin(abc.ABC):
```
Comment (konradko) [inline enrolment/views.py]: Mixin or abstract class, pick one :P

File: enrolment/templates/bidi/browse-companies.html (hunk 1 above)
Comment (richtier) [inline enrolment/templates/bidi/browse-companies.html]: `bidi` is indeed common short form. e.g.,: 

https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#get-current-language-bidi
https://github.com/django/django/blob/master/django/conf/locale/__init__.py#L36

File: enrolment/templates/govuk_layout.html (hunk 2 above)
Comment (richtier) [inline enrolment/templates/govuk_layout.html]: > LANGUAGE_BIDI

That's a commonly used variable name for indicating "is the current language bidirectional"?
https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#get-current-language-bidi

> `dir="rtl"`

It's a best practice for rtl language layout: https://www.w3.org/International/questions/qa-html-dir

File: enrolment/tests/test_views.py (hunk 3 above)
Comment (richtier) [inline enrolment/tests/test_views.py]: correct

File: enrolment/templates/govuk_layout.html (hunk 2 above)
Comment (konradko) [inline enrolment/templates/govuk_layout.html]: Good stuff

File: enrolment/tests/test_views.py (hunk 3 above)
Comment (konradko) [inline enrolment/tests/test_views.py]: assuming bidi is actually bidirectional, not just RTL

**Pull request merged**